src/barnetilsyn/BarnetilsynContext.tsx

```diff
@@ -95,18 +94,31 @@ const [BarnetilsynSøknadProvider, useBarnetilsynSøknad] = createUseContext(
     };
 
     const hentForrigeSøknadBarnetilsyn = async (): Promise<void> => {
-      return hentDataFraForrigeBarnetilsynSøknad().then(
-        (tidligereVersjon?: ForrigeSøknad) => {
-          if (tidligereVersjon) {
-            settSøknad((prevSøknad) => ({
-              ...prevSøknad,
-              ...tidligereVersjon,
-            }));
-          }
-        }
-      );
+      const forrigeSøknad = await hentDataFraForrigeBarnetilsynSøknad();
+      if (forrigeSøknad) {
+        settSøknad((prevSøknad) => ({
+          ...prevSøknad,
+          ...forrigeSøknad,
+          person: {
+            ...prevSøknad.person,
+            barn: [
+              ...forrigeSøknad.person.barn,
+              ...prevSøknad.person.barn.filter(
+                (barn) =>
+                  !forrigeSøknad.person.barn.some(
+                    (prevBarn) => prevBarn.ident.verdi === barn.ident.verdi
+                  )
+              ),
```

Comment:
Antar at dette er nye barn siden forrige søknad? 
Kan det trekkes ut som en variabel på utsiden av `settSøknad` for å bli litt mer lesbart? 🤔

Reply:
Enig i det. [Fikset](https://github.com/navikt/familie-ef-soknad/pull/1556/commits/c9133c21163a64d638948361e3f78bcd4fbf0719)